tests/unit/logic/vote.spec.js

```diff
@@ -0,0 +1,1012 @@
+					expect(callback.args[0][0]).to.include('Invalid vote at index')
+				})
+			})
+
+			context('if there are duplicate votes', function () {
+
+				beforeEach(function () {
+					clock = sinon.useFakeTimers()
+					Vote.__set__('setImmediate', setImmediate)
+					callback = sinon.spy()
+					trs = {asset: {votes: ['+b94d27b9934d3e08a52e52d7da7dabfac484efe37a5380ee9088f7ace2efcde9', '+b94d27b9934d3e08a52e52d7da7dabfac484efe37a5380ee9088f7ace2efcde9']}}
+					vote.verify(trs, sender, callback)
+				})
+
+				afterEach(function () {
+					clock.restore()
+				})
+
+				it('should return an \'Multiple votes for same delegate are not allowed\' error', function () {
+
+					clock.runAll()
+					expect(callback.called).to.be.true
+					expect(callback.args[0][0]).to.include('Multiple votes for same delegate are not allowed')
+				})
+			})
+
+			context('if everything is ok', function () {
+
+				beforeEach(function () {
+					clock = sinon.useFakeTimers()
+					Vote.__set__('setImmediate', setImmediate)
+					Vote.__set__('self.checkConfirmedDelegates', function (trs, cb) {
+						return setImmediate(cb)
+					})
+					callback = sinon.spy()
+					trs = {asset: {votes: ['+b94d27b9934d3e08a52e52d7da7dabfac484efe37a5380ee9088f7ace2efcde9', '+37b6d3fc89fb418fa5e3799e760cc2b1733a567daebb08a531d3d7e8b24f2d22']}}
+					vote.verify(trs, sender, callback)
+				})
+
+				afterEach(function () {
+					clock.restore()
+				})
+
+				it('should call to callback without errors', function () {
+
+					clock.runAll()
+					expect(callback.called).to.be.true
+					expect(callback.args[0][0]).to.equals(undefined)
+				})
+			})
+		})
+
+		describe('verifyVote()', function () {
+
+			var clock, callback
+
+			context('if vote is not a string', function () {
+
+				beforeEach(function () {
+					clock = sinon.useFakeTimers()
+					Vote.__set__('setImmediate', setImmediate)
+					callback = sinon.spy()
+				})
+
+				afterEach(function () {
+					clock.restore()
+				})
+
+				it('should return an \'Invalid vote type\' error', function () {
+					vote.verifyVote(123, callback)
+					clock.tick()
+					expect(callback.called).to.be.true
+					expect(callback.args[0][0]).to.equals('Invalid vote type')
+				})
+			})
+
+			context('if vote has not a valid format', function () {
+
+				beforeEach(function () {
+					clock = sinon.useFakeTimers()
+					Vote.__set__('setImmediate', setImmediate)
+					callback = sinon.spy()
+				})
+
+				afterEach(function () {
+					clock.restore()
+				})
+
+				it('should return an \'Invalid vote format\' error', function () {
+					vote.verifyVote('+b94d27b993-d3e08a52e52d7da7dabfac484efe37a5380ee9088f7ace2efcde9', callback)
+					clock.tick()
+					expect(callback.called).to.be.true
+					expect(callback.args[0][0]).to.equals('Invalid vote format')
+				})
+			})
+
+			context('if vote length is not 65', function () {
+
+				beforeEach(function () {
+					clock = sinon.useFakeTimers()
+					Vote.__set__('setImmediate', setImmediate)
```

Comment:
I see that you use context instead of sub `describe` while it's a safe guard we shouldn't misuse contexts especially if they're not really used.

Also it would be better to have one `beforeEach` and `afterEach` where you `sinon.useFakeTimers` and also do other shared stuff between tests (like the setImmediate thing?)

Right now there is a lot of boilerplate involving the init/teardown of test groups.